keanu-project/src/main/java/io/improbable/keanu/model/LogisticRegression.java

```diff
@@ -0,0 +1,111 @@
+package io.improbable.keanu.model;
+
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.probabilistic.BernoulliVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+public class LogisticRegression implements LinearModel {
```

Comment:
It doesn't seem right that this class is quite different to LinearRegression. Logistic regression is effectively just linear regression with a sigmoid function wrapped around it. Can we do some more code reuse between the two?

Reply:
Good spot. I've made the `Logistic` call out to `Linear` under the hood and perform the extra sigmoid op's at the end